app/controllers/api/v1/visualization/annotations_controller.rb

```diff
@@ -100,6 +100,10 @@ def index
 
         def show
           annotation = self.class.get_selected_annotation(@study, params)
+          if annotation.nil?
+            head :not_found and return
+          end
```

Comment:
Kudos on the key insight to prevent server resource flooding by failing fast here, and equivalent areas.